src/frontend/src/utils/iiConnection.ts

```diff
@@ -172,6 +164,26 @@ export class IIConnection {
   }
 
   // Create an actor representing the backend
+
+  static async createActor(
+    delegationIdentity: DelegationIdentity
+  ): Promise<ActorSubclass<_SERVICE>> {
+    const agent = new HttpAgent({ identity: delegationIdentity });
+    // try to recognize development environments
+    // this is conservative, and defaults to _not_ fetching the root key
+    if (
+      window.location.origin.startsWith("http://localhost:") ||
+      window.location.origin.endsWith(".dfinity.network")
+    ) {
+      await agent.fetchRootKey();
```

Comment:
Question: what's wrong with modifying `fetchRootKey` to first check the status endpoint of the replica it's talking to? If the root key is available (in case of local dev), then use that, otherwise, use the IC mainnet root key.

Reply:
Two reasons
 * we still _do_ serve the root key from the IC mainnet
 * even if we woudn’t, a MITM attacker could, and the whole thing breaks apart

Reply:
What stops a rogue replica from serving it's own root key on the mainnet?

Reply:
Nothing, that’s why we shouldn’t ever fetch the key in non-dev situations

Reply:
> we still do serve the root key from the IC mainnet

Is that something that we should change soon? It's a fairly easy change.

Reply:
Yes, but one that will break many applications; the changes to the agents hardcoding the IC root key are rather recent.